sklearn/decomposition/kernel_pca.py

```diff
@@ -295,8 +323,16 @@ def transform(self, X):
         """
         check_is_fitted(self, 'X_fit_')
 
+        # Compute centered gram matrix between X and training data X_fit_
         K = self._centerer.transform(self._get_kernel(X, self.X_fit_))
-        return np.dot(K, self.alphas_ / np.sqrt(self.lambdas_))
+
+        # scale eigenvectors (properly account for null-space for dot product)
+        nz = np.flatnonzero(self.lambdas_)
+        scaled_alphas = np.zeros(self.alphas_.shape)
+        scaled_alphas[:, nz] = self.alphas_[:, nz] / np.sqrt(self.lambdas_[nz])
```

Comment:
Is this a separate bug fix to the rest of the PR?

Reply:
yes this PR is the second of a series of 3. that particular bit of code comes from https://github.com/scikit-learn/scikit-learn/pull/12143 .